### PR TITLE
Additional fields for `VmmPingResponse`

### PR DESCRIPTION
**Pull request by brkp**
This PR introduces several additional fields to the `VmmPingResponse` struct beyond the `version` information. The updated response now includes the `build_version` and `pid` fields as shown below:
```json
{
  "build_version": "v31.0-4-g48084c33-dirty",
  "version": "31.0.0",
  "pid": 60639
}
```

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -17,8 +18,13 @@ fn main() {
...
-    // variable BUILT_VERSION, so that it can be reused from the binary.
+    // variables BUILD_VERSION and BUILD_TIMESTAMP, so that it can be reused
+    // from the binary.
     // Particularly, this is used from src/main.rs to display the exact
-    // version.
-    println!("cargo:rustc-env=BUILT_VERSION={version}");
+    // version and build timestamp.
+    println!("cargo:rustc-env=BUILD_VERSION={version}");
+    println!(
+        "cargo:rustc-env=BUILD_TIMESTAMP={}",
```
Comment (liuw) [inline build.rs:27]: I'm afraid this will break reproducible build.

What do you need this time stamp for?

Comment (lperkov) [reply build.rs:27]: We would like to know if there has been any modifications to the source code at build time. We could live without build_timestamp, but would it be possible to include the build hash (aka. build_version)?

Comment (liuw) [reply build.rs:27]: Using the commit hash is a much better indicator than a time stamp for your use case.

The hash is already there, isn't it?

Here is what I see from a build:

```
$ cloud-hypervisor --version
cloud-hypervisor v30.0-92-g38a1b45783a6
```

If the source tree has been modified (according to git), you get the `-dirty` suffix.

Comment (brkp) [reply build.rs]: > I'm afraid this will break reproducible build.

Thanks for pointing that out, I completely overlooked that possibility.

**Review by rbradford — APPROVED** [review]

**Review by likebreath — APPROVED** [review]

File: vmm/src/lib.rs [hunk 2]
```diff
@@ -390,6 +390,25 @@ struct VmMigrationConfig {
...
+    pub fn new(build_version: &str, version: &str) -> Self {
+        Self {
+            build_version: build_version.to_owned(),
+            version: version.to_owned(),
+        }
+    }
+
+    pub fn dummy() -> Self {
+        Self::new("dummy", "dummy")
+    }
```
Comment (likebreath) [inline vmm/src/lib.rs:409]: Considering it will only be used for unit test code, why don't you simply call `new()` and get rid of the `dummy` method?

**Review by likebreath — CHANGES_REQUESTED** [review]

File: vmm/src/lib.rs (hunk 2 above)
Comment (brkp) [reply vmm/src/lib.rs]: Makes perfect sense, thanks for pointing it out.

**Review by likebreath — APPROVED** [review]

**Pull request merged**